source/common/http/url_utility.cc

```diff
@@ -0,0 +1,95 @@
+#include "common/http/url_utility.h"
+
+#include <sys/types.h>
+
+#include <iostream>
+#include <string>
+
+#include "common/common/assert.h"
+#include "common/common/empty_string.h"
+#include "common/common/utility.h"
+
+#include "absl/strings/numbers.h"
+#include "absl/strings/str_cat.h"
+
+namespace Envoy {
+namespace Http {
+namespace Utility {
+
+bool Url::initialize(absl::string_view absolute_url, bool is_connect) {
+  // TODO(dio): When we have access to base::StringPiece, probably we can convert absolute_url to
+  // that instead.
+  GURL parsed(std::string{absolute_url});
+  if (is_connect) {
+    return initializeForConnect(std::move(parsed));
+  }
+
+  // TODO(dio): Apply stricter validity checking by taking account accepted schemes:
```

Comment:
Which of these do you want to land in this PR and which later? Should some tracking issues be opened if we're punting?

Reply:
I think I want to apply check for `SchemeIsHTTPOrHTTPS` in this patch since the call sites for this Today are only for HTTP/HTTPS. When it is needed, we can add another or criteria `SchemeIsWSOrWSS`. WDYT?